// @dcloudio/uni-app-uts/dist/plugins/ios/plugin.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.uniAppIOSPlugin = void 0;
const fs_extra_1 = __importDefault(require("fs-extra"));
const uni_cli_shared_1 = require("@dcloudio/uni-cli-shared");
const utils_1 = require("../utils");
const css_1 = require("./css");
function uniAppIOSPlugin() {
    const inputDir = process.env.UNI_INPUT_DIR;
    const outputDir = process.env.UNI_OUTPUT_DIR;
    // 开始编译时，清空输出目录
    function emptyOutDir() {
        if (fs_extra_1.default.existsSync(outputDir)) {
            (0, uni_cli_shared_1.emptyDir)(outputDir);
        }
    }
    emptyOutDir();
    return {
        name: 'uni:app-uts',
        apply: 'build',
        uni: (0, utils_1.createUniOptions)('ios'),
        config() {
            return {
                base: '/',
                build: {
                    emptyOutDir: false,
                    assetsInlineLimit: 0,
                    rollupOptions: {
                        input: (0, uni_cli_shared_1.resolveMainPathOnce)(inputDir),
                        external: ['vue', '@vue/shared'],
                        output: {
                            name: 'AppService',
                            banner: ``,
                            format: 'iife',
                            entryFileNames: uni_cli_shared_1.APP_SERVICE_FILENAME,
                            globals: {
                                vue: 'Vue',
                                '@vue/shared': 'uni.VueShared',
                            },
                        },
                    },
                },
            };
        },
        configResolved(config) {
            (0, utils_1.configResolved)(config);
            (0, uni_cli_shared_1.injectCssPlugin)(config);
            (0, uni_cli_shared_1.injectCssPostPlugin)(config, (0, css_1.uniAppCssPlugin)(config));
        },
    };
}
exports.uniAppIOSPlugin = uniAppIOSPlugin;
